src/main/scala/TTFI.scala

```diff
@@ -417,37 +361,37 @@ object TTFI {
               case _ => throw new Exception("Impossible happened!") // unify?
             }
           }
-          def fromTreeExt_[repr[+_]](self: Tree[Integer] => Either[ErrMsg, (ExpSym[repr] => MulSym[repr] => repr[Integer])])(x: Tree[Integer]): Either[ErrMsg, (ExpSym[repr] => MulSym[repr] => repr[Integer])] = ???
-          def fromTreeExt[repr[+_]](x: Tree[Integer]): Either[ErrMsg, (ExpSym[repr] => MulSym[repr] => repr[Integer])] =
+          def fromTreeExt_[repr[_]](self: Tree[Integer] => Either[ErrMsg, (ExpSym[repr] => MulSym[repr] => repr[Integer])])(x: Tree[Integer]): Either[ErrMsg, (ExpSym[repr] => MulSym[repr] => repr[Integer])] = ???
+          def fromTreeExt[repr[_]](x: Tree[Integer]): Either[ErrMsg, (ExpSym[repr] => MulSym[repr] => repr[Integer])] =
             Fix(fromTreeExt_[repr])(x)
         }
 
-        object Poly {
-          def fromTree_[repr[+_]](self: ((Tree[Integer], ExpSym[repr])) => Either[ErrMsg, repr[Integer]])(args: (Tree[Integer], ExpSym[repr])) = args match {
-            case (x, s1) => x match {
-              case Node("Lit", Seq(Leaf(x))) => safeRead(x).right.map(s1.lit(_))
-              case Node("Neg", Seq(x)) => self((x, s1)).right.map(s1.neg(_))
-              case Node("Add", Seq(x, y)) => for {
-                a <- self((x, s1)).right
-                b <- self((y, s1)).right
-              } yield s1.add(a)(b)
-              case _ => Left(s"Parse error in ${x}")
-            }
-          }
-          def fromTree[repr[+_]](x: Tree[Integer])(implicit s1: ExpSym[repr]): Either[ErrMsg, repr[Integer]] =
-            Fix(fromTree_[repr])((x, s1))
-
-          def fromTreeExt_[repr[+_]](self: ((Tree[Integer], MulSym[repr], ExpSym[repr])) => Either[ErrMsg, repr[Integer]])(args: (Tree[Integer], MulSym[repr], ExpSym[repr])): Either[ErrMsg, repr[Integer]] = args match {
-            case (x, s1, s2) => x match {
-              case Node("Mul", Seq(x, y)) => for {
-                a <- self((x, s1, s2)).right
-                b <- self((y, s1, s2)).right
-              } yield s1.mul(a)(b)
-              case x => fromTree_((x: (Tree[Integer], ExpSym[repr])) => self(x._1, s1, x._2))((x, s2))
-            }
-          }
-          def fromTreeExt[repr[+_]](x: Tree[Integer])(implicit s1: MulSym[repr], s2: ExpSym[repr]): Either[ErrMsg, repr[Integer]] = Fix(fromTreeExt_[repr])((x, s1, s2))
-        }
+        // object Poly {
```

Comment:
This part does not compile anymore. @suhailshergill what version of scala was used initially? Maybe this is because of the compiler options.